service/collector/cluster_info.go

```diff
@@ -21,7 +23,7 @@ var clustersDesc *prometheus.Desc = prometheus.NewDesc(
 	nil,
 )
 
-func (c *Collector) collectClusterInfo(ch chan<- prometheus.Metric) {
+func (c *Collector) collectClusterInfo(ch chan<- prometheus.Metric, clients []aws.Clients) {
```

Comment:
I am not super happy about passing those `aws.Clients` around in every function. This would need some refactoring and moving every collector under its own package would also help here.

Reply:
I think splitting the collectors into separate packages would be a useful refactor. Doesn't have to be in this PR though IMO.

Reply:
I'm curious how moving to a separate package can help here?

Reply:
If we are talking about separate objects I agree with assumption a new instance is created on every call. If we are talking about packages I still don't understand.

Reply:
Yes we are talking about separate object, and also moving those objects into their own package for even more clarity (e.g. `service/collector/vpc`, `service/collector/trusted_advisor`).

Reply:
Splitting into separate objects makes sense to me and I like that package structure. But yes not needed in this PR.

Reply:
My take on that is visibility is more important than the imports themselves. I can take more information from `collector.VPC` than from `vpc.VPC`. The latter can be anything.

Reply:
This is a good point. I agree that `collector.VPC` is more visible when reading.